File: DropDownList/CustomItemWithMultipleImages/RadListControlCustomItemsCS/Helpers/CustomListVisualItem.cs
using System;
using System.Windows.Forms;
using Telerik.WinControls;
using Telerik.WinControls.Enumerations;
using Telerik.WinControls.Layouts;
using Telerik.WinControls.Primitives;
using Telerik.WinControls.UI;

namespace RadListControlCustomItems.Helpers
{
    /// <summary>
    /// The custom list visual item.
    /// </summary>
    public class CustomListVisualItem : RadListVisualItem
    {
        #region Constants and Fields

        /// <summary>
        /// The image.
        /// </summary>
        private readonly ImagePrimitive image = new ImagePrimitive();

        /// <summary>
        /// The image 2.
        /// </summary>
        private readonly ImagePrimitive image2 = new ImagePrimitive();

        /// <summary>
        /// The label.
        /// </summary>
        private readonly RadLabelElement label = new RadLabelElement();

        #endregion Constants and Fields

        #region Constructors and Destructors

        /// <summary>
        /// Initializes static members of the <see cref="CustomListVisualItem"/> class.
        /// </summary>
        static CustomListVisualItem()
        {
            SynchronizationProperties.Add(CustomListDataItem.ImageProperty);
            SynchronizationProperties.Add(CustomListDataItem.Image2Property);
            SynchronizationProperties.Add(CustomListDataItem.NameProperty);
        }

        #endregion Constructors and Destructors

        #region Properties

        /// <summary>
        /// Gets ThemeEffectiveType.
        /// </summary>
        protected override Type ThemeEffectiveType
        {
            get
            {
                return typeof(RadListVisualItem);
            }
        }

        #endregion Properties

        #region Methods

        /// <summary>
        /// The create child elements.
        /// </summary>
        protected override void CreateChildElements()
        {
            base.CreateChildElements();

            this.label.StretchHorizontally = true;

            var stack = new StackLayoutPanel { Orientation = Orientation.Horizontal };

            this.image.ImageScaling = ImageScaling.SizeToFit;
            this.image2.ImageScaling = ImageScaling.SizeToFit;

            stack.Children.Add(this.label);
            stack.Children.Add(this.image);
            stack.Children.Add(this.image2);

            this.Children.Add(stack);
        }

        /// <summary>
        /// The property synchronized.
        /// </summary>
        /// <param name="property">
        /// The property.
        /// </param>
        protected override void PropertySynchronized(RadProperty property)
        {
            var dataItem = (CustomListDataItem)this.Data;

            this.image2.Image = dataItem.Image2;
            this.image.Image = dataItem.Image;
            this.label.Text = dataItem.Name;
        }

        #endregion Methods
    }
}

File: DropDownList/CustomItemWithMultipleImages/RadListControlCustomItemsVB/Helpers/CustomListVisualItem.vb
Imports Telerik.WinControls.UI
Imports Telerik.WinControls.Primitives
Imports Telerik.WinControls.Layouts
Imports Telerik.WinControls.Enumerations
Imports Telerik.WinControls

''' <summary>
''' The custom list visual item.
''' </summary>
Public Class CustomListVisualItem
    Inherits RadListVisualItem
#Region "Constants and Fields"

    ''' <summary>
    ''' The label.
    ''' </summary>
    Private label As RadLabelElement

    ''' <summary>
    ''' The image.
    ''' </summary>
    Private Shadows image As ImagePrimitive

    ''' <summary>
    ''' The image 2.
    ''' </summary>
    Private image2 As ImagePrimitive

#End Region

#Region "Constructors and Destructors"

    ''' <summary>
    ''' Initializes static members of the <see cref="CustomListVisualItem"/> class.
    ''' </summary>
    Shared Sub New()
        SynchronizationProperties.Add(CustomListDataItem.ImageProperty)
        SynchronizationProperties.Add(CustomListDataItem.Image2Property)
        SynchronizationProperties.Add(CustomListDataItem.NameProperty)
    End Sub

#End Region

#Region "Properties"

    ''' <summary>
    ''' Gets ThemeEffectiveType.
    ''' </summary>
    Protected Overrides ReadOnly Property ThemeEffectiveType() As Type
        Get
            Return GetType(RadListVisualItem)
        End Get
    End Property

#End Region

#Region "Methods"

    ''' <summary>
    ''' The create child elements.
    ''' </summary>
    Protected Overrides Sub CreateChildElements()
        MyBase.CreateChildElements()

        label = New RadLabelElement()
        image = New ImagePrimitive()
        image2 = New ImagePrimitive()

        Me.label.StretchHorizontally = True

        Dim stack = New StackLayoutPanel() With { _
         .Orientation = Orientation.Horizontal _
        }

        Me.image.ImageScaling = ImageScaling.SizeToFit
        Me.image2.ImageScaling = ImageScaling.SizeToFit

        stack.Children.Add(Me.label)
        stack.Children.Add(Me.image)
        stack.Children.Add(Me.image2)

        Me.Children.Add(stack)
    End Sub

    ''' <summary>
    ''' The property synchronized.
    ''' </summary>
    ''' <param name="property">
    ''' The property.
    ''' </param>
    Protected Overrides Sub PropertySynchronized(ByVal [property] As RadProperty)
        Dim dataItem = DirectCast(Me.Data, CustomListDataItem)
        Dim boItem = TryCast(dataItem.DataBoundItem, BussinessObject)

        If boItem Is Nothing Then
            Return
        End If

        Me.image2.Image = boItem.Image2
        Me.image.Image = boItem.Image
        Me.label.Text = boItem.Name
    End Sub

#End Region
End Class